src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,66 @@
+package duke;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+
+/**
+ * Represents the list of tasks tracked by Duke.
+ */
+public class TaskList implements Serializable {
+    private final ArrayList<Task> tasks;
+
+    /**
+     * Class constructor which initialises a blank TaskList.
+     */
+    public TaskList() {
+        tasks = new ArrayList<>();
+    }
+
+    /**
+     * Adds a task to the list.
+     *
+     * @param task Task to be added
+     */
+    public void add(Task task) {
+        tasks.add(task);
+    }
+
+    /**
+     * Deletes the task at the specified index from the list, and returns the deleted task.
+     *
+     * @param index Index of the task in the list of tasks
+     * @return The task that has been deleted
+     */
+    public Task delete(int index) {
+        return tasks.remove(index);
+    }
+
+    /**
+     * Returns the number of tasks in the task list.
+     *
+     * @return Number of tasks in list
+     */
+    public int size() {
```

Comment:
Maybe the method can be named after a verb more such as getSize?

Reply:
I'm unsure about this, because the method for ArrayList is named 'size'.